austation/code/modules/mining/minebot.dm

```diff
@@ -0,0 +1,122 @@
+#define MINEDRONE_COLLECT 1
+#define MINEDRONE_ATTACK 2
+
+/mob/living/simple_animal/hostile/mining_drone
+	var/beacons = 15  //maximum beacons that any one drone can drop
+	var/emagged = 0  //lets minebot shoot sentient life
+	var/obj/item/t_scanner/adv_mining_scanner/lesser/scanner //there is no code to turn this scanner off.
+	weather_immunities = list("ash")
+
+/mob/living/simple_animal/hostile/mining_drone/Initialize()
+	. = ..()
+	stored_gun.overheat_time = 10
+
+	scanner = new(src)
+	scanner.toggle_on()
+	var/datum/action/innate/minedrone/marker_beacon/beacon_action = new()
+	beacon_action.Grant(src)
+
+	if(emagged)
+		to_chat(src, "<span class='danger'>SYSTEM IRREGULARITIES DETECTED</span>")
+		to_chat(src, "<span class='warning'>Directives offline \nSeek additional instructions from the nearest <b>SYNDICATE</b> personel.</span>")
+
+/mob/living/simple_animal/hostile/mining_drone/emag_act()
+	..()
+	emagged = 1
+	to_chat(src, "<span class='danger'>SYSTEM IRREGULARITIES DETECTED</span>")
+	to_chat(src, "<span class='warning'>Directives offline \nSeek additional instructions from the nearest <b>SYNDICATE</b> personel.</span>")
+
+obj/item/gun/energy/kinetic_accelerator/minebot/afterattack(atom/target, mob/living/user, flag, params)
+	if(istype(target, /mob/living) && istype(user, /mob/living/simple_animal/hostile/mining_drone))  //Are we a minebot, firing at a mob
+		var/mob/living/M = target
+		var/mob/living/simple_animal/hostile/mining_drone/H = user
+		if((M.ckey) && (!H.emagged))  //Are we firing at sentient life without permission
+			to_chat(user, "<span class='warning'>Invalid target. \n Target is sentient. \n engaging weapon lockdown.</span>")
+			H.mode = MINEDRONE_COLLECT  //let's turn off the guns, just to be safe
+			return
+	..()
+
+/datum/action/innate/minedrone/marker_beacon
+	name = "Drop Marker"
+	button_icon_state = "mech_zoom_off"
+
+/datum/action/innate/minedrone/marker_beacon/Activate()
+	if (!istype(owner, /mob/living/simple_animal/hostile/mining_drone))  //Are we a minebot
+		return
+	var/mob/living/simple_animal/hostile/mining_drone/M = owner
+	if (!M.beacons)  //Got any beacons
+		return
+	M.beacons--
+	var/obj/structure/marker_beacon/B = new(get_turf(owner))
+	to_chat(M, "<span class='notice'>You place a [B].</span>")
+	to_chat(M, "<span class='notice'>You have [M.beacons] beacons remaining.</span>")
```

Comment:
POV: you still can't pickup beacons to recycle your supply

Reply:
now you can